src/Nethermind/Nethermind.Blockchain.Test/BlockhashProviderTests.cs

```diff
@@ -277,14 +277,14 @@ public void Eip2935_init_block_history_and_then_get_hash(int chainLength)
             result = provider.GetBlockhash(current.Header, chainLength);
             Assert.That(result, Is.EqualTo(head?.Hash));
 
-            AssertGenesisHash(provider, current.Header, genesis.Hash!);
+            AssertGenesisHash(Prague.Instance, provider, current.Header, genesis.Hash);
         }
 
-        private static void AssertGenesisHash(BlockhashProvider provider, BlockHeader currentHeader,
-            Hash256 genesisHash)
+        private static void AssertGenesisHash(IReleaseSpec spec, BlockhashProvider provider, BlockHeader currentHeader,
+            Hash256? genesisHash)
         {
             Hash256? result = provider.GetBlockhash(currentHeader, 0);
-            if (currentHeader.Number > Eip2935Constants.RingBufferSize)
+            if (currentHeader.Number > Eip2935Constants.RingBufferSize || spec.IsEip7709Enabled || currentHeader.Number > 256)
```

Comment:
this condition does not make sense - i think this should be
`(spec.IsEip7709Enabled && currentHeader.Number > Eip2935Constants.RingBufferSize) || currentHeader.Number > 256`